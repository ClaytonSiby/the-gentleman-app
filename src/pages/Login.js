import React, { useRef } from 'react';
import { useDispatch } from 'react-redux';
import { Form } from 'react-bootstrap';
import requestToLogin from '../redux/users/login/loginActions';
import styles from '../assets/scss/login.module.scss';

const Login = () => {
  const email = useRef(null);
  const password = useRef(null);
  const dispatch = useDispatch();

  const handleLogin = (e) => {
    e.preventDefault();

    const userCredentials = {
      email: email.current.value,
      password: password.current.value,
    };

    dispatch(requestToLogin(userCredentials));
  };

  return (
    <div className={`${styles.login}`}>
      <h2 className="mb-3">Login</h2>
      <Form onSubmit={handleLogin}>
        <Form.Group controlId="formBasicEmail">
          <Form.Label>Email address</Form.Label>
          <Form.Control ref={email} type="email" placeholder="Enter email" />
          <Form.Text className="text-muted">
            We&apos;ll never share your email with anyone else.
          </Form.Text>
        </Form.Group>

        <Form.Group controlId="formBasicPassword">
          <Form.Label>Password</Form.Label>
          <Form.Control ref={password} type="password" placeholder="Password" />
        </Form.Group>
        <button className="btn btn-default border bg-light my-3" type="submit">
          Login
        </button>
      </Form>
    </div>
  );
};

export default Login;
