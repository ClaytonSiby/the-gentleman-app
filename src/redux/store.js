import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { persistStore } from 'redux-persist';
import persistReducer from './rootReducer';

const store = createStore(
  persistReducer,
  composeWithDevTools(applyMiddleware(thunk)),
);

const persistor = persistStore(store);

export { store, persistor };
