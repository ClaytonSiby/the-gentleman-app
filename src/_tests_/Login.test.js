import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import Login from '../pages/Login';
import store from './Suits.test';

describe('Login', () => {
  it('Renders the Login page successfully', () => {
    render(
      <Provider store={store}>
        <Login />
      </Provider>,
    );
  });
});
