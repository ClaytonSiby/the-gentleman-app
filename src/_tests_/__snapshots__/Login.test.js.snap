// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Renders consistently to the DOM 1`] = `
<div
  className="login"
>
  <h2
    className="mb-3"
  >
    Login
  </h2>
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="formBasicEmail"
      >
        Email address
      </label>
      <input
        className="form-control"
        id="formBasicEmail"
        placeholder="Enter email"
        type="email"
      />
      <small
        className="text-muted form-text"
      >
        We'll never share your email with anyone else.
      </small>
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="formBasicPassword"
      >
        Password
      </label>
      <input
        className="form-control"
        id="formBasicPassword"
        placeholder="Password"
        type="password"
      />
    </div>
    <button
      className="btn btn-default border bg-light my-3"
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;

exports[`Suits renders consistently to the DOM 1`] = `
<div
  className="text-light"
>
  <div
    className="loading"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\">
  <path transform=\\"translate(2)\\" d=\\"M0 12 V20 H4 V12z\\"> 
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\"  />
  </path>
  <path transform=\\"translate(8)\\" d=\\"M0 12 V20 H4 V12z\\">
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.2\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\"  />
  </path>
  <path transform=\\"translate(14)\\" d=\\"M0 12 V20 H4 V12z\\">
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.4\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\" />
  </path>
  <path transform=\\"translate(20)\\" d=\\"M0 12 V20 H4 V12z\\">
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.6\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\" />
  </path>
  <path transform=\\"translate(26)\\" d=\\"M0 12 V20 H4 V12z\\">
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.8\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\" />
  </path>
</svg>
",
        }
      }
      style={
        Object {
          "fill": "grey",
          "height": 70,
          "width": 80,
        }
      }
    />
  </div>
</div>
`;
